compiler/src/dotty/tools/dotc/transform/ExplicitOuter.scala

```diff
@@ -81,14 +81,19 @@ class ExplicitOuter extends MiniPhaseTransform with InfoTransformer { thisTransf
         !needsOuterAlways(cls) &&
         impl.existsSubTree(referencesOuter(cls, _)))
       ensureOuterAccessors(cls)
-    if (hasOuter(cls)) {
+
+    val hasOuterFlag = hasOuter(cls)
```

Comment:
Can we rename `hasOuterFlag` to `clsHasOuter`, or something like this?

I was reading `hasOuterFlag` as "has the Outer flag set" but of course there is no such flag.
Otherwise LGTM.